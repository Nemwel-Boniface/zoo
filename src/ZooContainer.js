import React, { useEffect } from 'react';
import { Route, Routes } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Details from './components/details';
import Header from './components/header';
import Landing from './components/landing';
import Zoo from './components/Zoo';
import { getAnimalsFromAPI } from './redux/Zoo/Zoo';

function ZooContainer() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getAnimalsFromAPI());
  }, []);
  return (
    <div className="ZooContainer">
      <Header />
      <Landing />
      <Routes>
        <Route exact path="/" element={<Zoo />} />
        <Route exact path="/details/:id" element={<Details />} />
      </Routes>
    </div>
  );
}

export default ZooContainer;
