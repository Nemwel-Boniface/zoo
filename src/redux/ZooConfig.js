import { createStore,  applyMiddleware } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import ZooReducer from "./Zoo/Zoo"

const middleWares = [thunk, logger ]

const store = createStore(
  ZooReducer,
  applyMiddleware(...middleWares)
)
export default store;