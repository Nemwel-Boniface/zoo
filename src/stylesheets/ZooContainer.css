* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Lato', sans-serif;
  color: #fff;
}

.ZooContainer {
  width: 100%;
  background: rgba(67, 105, 178, 0.8);
}
