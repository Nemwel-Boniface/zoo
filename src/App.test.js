import { createStore, applyMiddleware, combineReducers } from 'redux';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import ZooReducer from './redux/Zoo/Zoo';

const zoored = combineReducers({ animals: ZooReducer });
const middleWares = [thunk, logger];

const store = createStore(
  zoored,
  applyMiddleware(...middleWares),
);

afterEach(cleanup);
import ZooContainer from './ZooContainer';

describe('Test for the Zoo component', () => {
  test('Check whether the zoo page renders correctly', () => {
    const zoocont = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <ZooContainer />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(zoocont).toMatchSnapshot();
  });
});
