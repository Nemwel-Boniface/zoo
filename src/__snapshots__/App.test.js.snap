// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Zoo component Check whether the zoo page renders correctly 1`] = `
<div
  className="ZooContainer"
>
  <div
    className="Header"
  >
    <div
      className="logo"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="fa fa-arrow-left"
        />
        Home
      </button>
    </div>
    <div
      className="zooname"
    >
      <h2>
        Zoo
      </h2>
    </div>
    <div
      className="headerIcons"
    >
      <span>
        <i
          aria-hidden="true"
          className="fa fa-microphone"
        />
      </span>
      <span>
        <i
          aria-hidden="true"
          className="fa fa-cog"
        />
      </span>
    </div>
  </div>
  <div
    className="Zoo"
  >
    <div
      className="Landing"
    >
      <img
        alt="Gif"
        src="big5.png"
      />
      <div
        className="LandingDetails"
      >
        <h2>
          The Big 5
        </h2>
        <p>
          The big 5 can be found in Kenya's Maasai Mara park.
        </p>
      </div>
    </div>
    <div
      className="searchbar"
    >
      <input
        onChange={[Function]}
        placeholder="Search eg Lion..."
        type="text"
      />
    </div>
    <ul
      className="zooanimals"
    />
  </div>
</div>
`;
